server/aws-lsp-identity/src/utils/duckTyper.ts

```diff
@@ -0,0 +1,56 @@
+export class DuckTyper {
+    private static requirePropertyRule = 'requireProperty'
+    private static optionalPropertyRule = 'optionalProperty'
+    private static disallowPropertyRule = 'disallowProperty'
+
+    private rules: Record<string, string> = {}
+
+    requireProperty(name: string): DuckTyper {
+        this.rules[name] = DuckTyper.requirePropertyRule
+        return this
+    }
+
+    // Rule is only useful when onlyDefined === true
+    optionalProperty(name: string): DuckTyper {
+        this.rules[name] = DuckTyper.optionalPropertyRule
+        return this
+    }
+
+    disallowProperty(name: string): DuckTyper {
+        this.rules[name] = DuckTyper.disallowPropertyRule
+        return this
+    }
+
+    eval(value: object, onlyDefined: boolean = false): boolean {
```

Comment:
What does `onlyDefined` mean? Why is it needed? Maybe worth putting comment with explanation.